docs/arch/adr-010-new-versioning-schema.md

```diff
@@ -0,0 +1,145 @@
+
+To help us better comprehend the restrictions and limitations involved in upgrading between Channels, let us consider a hypothetical scenario. This will provide us with a practical example to illustrate the concept and highlight the challenges that may arise during the upgrading process.
+
+| Channel | Version|
+|---------|--------|
+| 1.25    | 1.0.0  |
+| 1.25    | 1.1.0  |
+| 1.25    | 1.2.0  |
+| 1.26    | 1.0.0  |
+| 1.26    | 1.1.0  |
+| 1.26    | 1.2.0  |
+| 1.27    | 1.0.0  |
+
+Based on the above-mentioned scenarios, the following table outlines various upgrade attempts and their respective outcomes:
+
+| From       | To        | Outcome |
+|------------|-----------|---------|
+|v1.25+1.0.0.|v1.25+1.2.0|Success. Upgrades inside the same Channel are allowed. |
+|v1.25+1.0.0.|v1.26+1.0.0|Failure. Users can only upgrade to the next Channel when they are already running the latest version of the current Channel. |
+|v1.25+1.2.0.|v1.26+1.2.0|Success. Upgrades from the latest version of the current Channel to any version on the next Channel are allowed. |
+|v1.25+1.2.0.|v1.27+1.0.0|Failure. Upgrades skipping a Channel are not allowed. |
+
+_Downgrades are strictly prohibited under any circumstances. Once a cluster is upgraded to a newer version in a specific Channel, there is no going back to a previous version. This is to maintain consistency and avoid any potential issues that may arise from incompatible configurations or data._
+
+### Additional Details
+
+#### Allowing for some degree of customisation
+Our aim is to maintain a level of opinionated approach towards the technology we offer, while also allowing a certain level of customisation for our customers. We believe that the customers should have the freedom to decide what they want to install on their clusters, and this includes factors such as having or not having an object storage API for example. In order to facilitate this, we plan to abstract such complex concepts into simpler and easier-to-understand terms.
+
+Let's exemplify: to make an Object Storage API available inside the cluster, customers can choose to install the "Minio" add-on. The details about which tool provides the Object Storage API inside kURL are not crucial for the users. What matters to them is having an ObjectStorageAPI available on their cluster. With this in mind, users can easily achieve their desired installation by using the following Installer:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+  version:  "1.8.0"
+  objectStorage:
+    enabled: true
+    credentialsSecrets:
```

Comment:
i am not familiar with this feature. is this an actual proposed addition to the spec?

Reply:
Yet another case where I should have restricted myself to the scope of the task at hand. Sorry for that. This is, as I stated a few lines after this, and "example" on how we could manage add-on configurations in the installer. This configuration only exists on my head and such a feature would, in my head, be implemented in our operator.